logic/dapp.js

```diff
@@ -100,8 +100,8 @@ DApp.prototype.verify = function(transaction, sender, cb, tx) {
 		return setImmediate(cb, 'Invalid application category');
 	}
 
-	var foundCategory = false;
-	for (i in dappCategories) {
+	let foundCategory = false;
+	for (const i in dappCategories) {
```

Comment:
You almost always want `of` rather than `in`. Is there a good reason for this?

Reply:
@willclarktech it was an existing implementation, so I have tried to alter it. however, in this scenario the `dappCategories` is an object I believe, so I think it should be fine.
Here is some reference https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...inf
Same goes for the below comment as well.

Reply:
From the docs (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#Difference_between_for...of_and_for...in):
> The for...in statement iterates over the enumerable properties of an object, in original insertion order.
The for...of statement iterates over data that iterable object defines to be iterated over.

That is, `in` lets you iterate over *all* properties, even the ones you don't intend people to iterate over.

Reply:
@willclarktech that's a good point, now I need to investigate each for loop implemented with `for in` and see if the intention and use the `for in` or `for of` accordingly.

Reply:
> That is, `in` lets you iterate over all properties, even the ones you don't intend people to iterate over.

When you want certain properties not to be iterated over, you set those as symbols. Then `in` won't see them. That's how it is done from ES6 onward.

Reply:
At any rate, best practice is to use `for ... of` unless you have some specific reason not to.